### PR TITLE
 containers: Add an example bastion container 

### PR DESCRIPTION
**Pull request by petervo**
* [x] image-refresh fedora-26

### CODE REVIEW

File: containers/bastion/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+Bastion Container
+====================
+
+This container provides an example of a container that can be used as a bastion to connect via ssh to remote servers that do not have the cockpit webservice installed or exposed. The target server is expected to at minimum have the ```cockpit-system``` package installed. This container can be run in two modes.
+
+By default on login cockpit will attempt to establish a ssh connection to the given host using the username and password provided.
+
+You can also mount an encrypted private key inside the container and set the environment variable ```COCKPIT_SSH_KEY_PATH``` to point to it.
+
+```docker run -e COCKPIT_SSH_KEY_PATH='/var/secret/id_rsa' -v ~/.ssh/id_rso:/var/secret/id_rsa cockpit/bastion```
```
Comment (stefwalter) [inline containers/bastion/README.md]: The above command has a typo ```rso```

Comment (stefwalter) [reply containers/bastion/README.md]: Could we make this a location similar to the file below? Perhaps ```/secret/id_rsa``` or ```/var/secret/id_rsa```. If the file is present it is used? I'd like this to be simpler to use and less repetitive. 

Eventually we can also make it be possible to have a directory of keys, each of which is tried.

File: containers/bastion/cockpit-auth-ssh-key [hunk 2]
```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env python
```
Comment (stefwalter) [inline containers/bastion/cockpit-auth-ssh-key]: Is it easy to make this python3? That way we don't have to port later. If not, no worries.

Comment (stefwalter) [inline containers/bastion/cockpit-auth-ssh-key]: Does this file need a license header?

File: containers/bastion/cockpit-auth-ssh-key [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+import base64
+import subprocess
+
+COCKPIT_SSH_COMMAND = "/usr/libexec/cockpit-ssh"
+
+def usage():
+    sys.stderr.write("usage {} [user@]host[:port]\n".format(sys.argv[0]))
+    sys.exit(os.EX_USAGE)
+
+def send_auth_command(challenge, response):
```
Comment (stefwalter) [inline containers/bastion/cockpit-auth-ssh-key]: Could you add a comment  near the top of the file with a link to protocol.md describing what we're w riting on stdout and reading from stdin?

File: containers/bastion/cockpit-auth-ssh-key [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+    assert response
+    assert response.startswith(starts), response
+    
+    val = base64.b64decode(response[len(starts):])
+    user, password = val.split(':', 1)
+    return user, password
+
+def send_decrypted_key(fname, password):
+    r, w = os.pipe()
+    p = subprocess.Popen(["openssl", "rsa", "-in", fname, "-passin", "fd:%i" % r],
```
Comment (stefwalter) [inline containers/bastion/cockpit-auth-ssh-key]: installer.sh or the Dockerfile needs to make sure openssl is installed, no?

**Comment by stefwalter** [comment]
This looks great! Nice documentation and work. just a few comments and request for a test.

File: containers/bastion/README.md (hunk 1 above)
Comment (petervo) [reply containers/bastion/README.md]: The reason I didn't do this is that there is some secrets automounting that happens on some systems (ei /usr/share/rhel/secrets on rhel/fedora systems) that depending on how things are configured might end up conflicting with cause causing the container to try with a key when it shouldn't. We could chose a really non standard place or continue to require the env var. WDYT?

File: containers/bastion/cockpit-auth-ssh-key (hunk 3 above)
Comment (petervo) [reply containers/bastion/cockpit-auth-ssh-key]: Added

**Comment by petervo** [comment]
Made changes except for changing the use of COCKPIT_SSH_KEY_PATH. I figured that can be a follow up if needed.

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tasks-17wnw.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-8387-20180110-193232/

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-tasks-b6lvv.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-8387-20180110-190606/

**Comment by cockpituous** [comment]
image-refresh fedora-26 done: https://github.com/cockpituous/cockpit/commits/image-refresh-fedora-26-20180110-194545

**Comment by stefwalter** [comment]
Who's gonna record a demo video for this container ... and update the release scripts for that container.

**Pull request closed without merging**